Make CookbookBundle work with eZ Platform

alongosz opened the pull request:
To showcase something for the Doc Team I tried to use this bundle and failed :) 
...so here there is the fix and README improvement.

- [x] Renamed in dacb1d5 `CreateXMLContentCommand` due to case mismatch (Symfony cache warmup crashed).
- [x] Based on my experience today I decided to improve `README.md` with a quick guide for a new developer to start "playing" with the bundle (86744ef). Hope you'll like it.

The remaining thing to do (separate PR) is maybe to introduce `.php_cs` config for fixer, however there is a lot of changes, so I'm not sure (I took conf from ezplatform and fixer changed 23 files out of 24 in the bundle).

alongosz commented:
Review ping @andrerom @bdunogier

bdunogier commented:
> followups for later:

I'd _almost_ suggest that we create tasks on the github issue tracker... it feels a bit overkill to have all of those on JIRA individually. And maybe have one JIRA that links to those ?

It would be easier to keep track of those in a simple place. Those tend to get lost in the volume on JIRA.

andrerom commented:
I agree, on repo's not part of the product we can easily use gh issues instead.

alongosz commented:
I've just found out that bundle service config is not loaded at all via DI (till now wasn't needed). It's due to bundle extension class naming.
Put on hold merge for now :)

andrerom commented:
Please get rid of the $name hack here and use normal symfony bundle name convention, it's not fully supported by Symfony as apparent by breaks in earlier Symfony 3.x versions and they might break this in 4.x for what we now.

alongosz commented:
> Please get rid of the $name hack here and use normal symfony bundle name convention

True, not really even needed as it is the same as bundle class name. 
With few other name and case fixes I got it working. Before pushing though I'm trying to finish the actual thing I was working on to make sure everything what is needed has been fixed here.

alongosz commented:
All done here: fixed extension class name and `$name` inside bundle class. I also noticed that I forgot to add to README the step about enabling this bundle, so this is updated as well.

alongosz commented:
@andrerom @bdunogier 
I rebased this and squashed fixups. I also found one missing `use` statement in the code for `find_fulltext` command. Fixed in ea3b049. The rest of the code remains the same as when reviewed.

I also created issues on GitHub as agreed a few comments earlier.

@andrerom 
> rename command

I don't actually remember which command you had in mind :)

andrerom commented:
> I don't actually remember which command you had in mind :)

Neither do I, might have been "ezpublish => ezplatform" or something.
